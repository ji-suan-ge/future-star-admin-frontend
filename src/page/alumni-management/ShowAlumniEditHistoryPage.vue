<template>
  <el-container>
    <el-main>
      <el-page-header @back="goBack()" content="资料详情"/>
      <el-divider/>
      <el-table
        class="history-table"
        :data="tableData"
        stripe>
        <el-table-column
          prop="id"
          align="center"
          label="编号">
        </el-table-column>
        <el-table-column
          prop="time"
          align="center"
          label="时间">
        </el-table-column>
        <!--        <el-table-column-->
        <!--          align="center"-->
        <!--          label="操作">-->
        <!--            <el-button-->
        <!--              type="primary"-->
        <!--              size="small"-->
        <!--              icon="el-icon-more"-->
        <!--              @click="showAllInfo()"-->
        <!--              circle>-->
        <!--            </el-button>-->
        <!--        </el-table-column>-->
      </el-table>
    </el-main>
    <el-footer>
      <el-pagination
        class="pagination"
        @current-change="handleCurrentChange"
        :current-page="currentPage"
        layout="prev, pager, next"
        :page-count="totalPage">
      </el-pagination>
    </el-footer>
  </el-container>
</template>

<script>
export default {
  name: 'ShowAlumniEditHistoryPage',
  data () {
    return {
      currentPage: 1,
      totalPage: 1,
      tableData: [
        {
          id: '1',
          time: '2020年1月3日14:36:18'
        },
        {
          id: '2',
          time: '2020年1月3日14:36:24'
        },
        {
          id: '3',
          time: '2020年1月3日14:36:30'
        },
        {
          id: '4',
          time: '2020年1月3日14:36:37'
        },
        {
          id: '5',
          time: '2020年1月3日14:36:41'
        },
        {
          id: '6',
          time: '2020年1月3日14:37:13'
        },
        {
          id: '7',
          time: '2020年1月3日14:37:21'
        },
        {
          id: '8',
          time: '2020年1月3日14:37:41'
        },
        {
          id: '9',
          time: '2020年1月3日14:38:02'
        },
        {
          id: '10',
          time: '2020年1月10日19:31:29'
        }
      ]
    }
  },
  methods: {
    goBack: function () {
      this.$router.go(-1)
    },
    handleCurrentChange (currentPage) {
      this.currentPage = currentPage
    }
  }
}
</script>

<style scoped>
  .history-table {
    width: 90%;
    margin: auto;
  }

  .el-pagination {
    text-align: center;
  }
</style>
