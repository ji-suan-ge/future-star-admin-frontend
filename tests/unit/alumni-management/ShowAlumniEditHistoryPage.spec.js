import { mount } from '@vue/test-utils'
import ShowAlumniEditHistoryPage from '../../../src/page/alumni-management/ShowAlumniEditHistoryPage'

describe('ShowAlumniEditHistoryPage', () => {
  const wrapper = mount(ShowAlumniEditHistoryPage)
  test('快照测试', async () => {
    await expect(wrapper.html()).toMatchSnapshot()
  })
  test('是一个 Vue 实例', async () => {
    await expect(wrapper.isVueInstance()).toBeTruthy()
  })
  test('按钮', async () => {
    const header = wrapper.find('el-page-header')
    await expect(header.attributes('content')).toBe('资料详情')
  })
})
