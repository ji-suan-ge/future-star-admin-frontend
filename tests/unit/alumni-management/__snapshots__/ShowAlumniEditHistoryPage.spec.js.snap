// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowAlumniEditHistoryPage 快照测试 1`] = `
<el-container>
  <el-main>
    <el-page-header content="资料详情"></el-page-header>
    <el-divider></el-divider>
    <el-table data="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" stripe="" class="history-table">
      <el-table-column prop="id" align="center" label="编号"></el-table-column>
      <el-table-column prop="time" align="center" label="时间"></el-table-column>
      <el-table-column align="center" label="操作">
        <el-button type="primary" size="small" icon="el-icon-more" circle=""></el-button>
      </el-table-column>
    </el-table>
  </el-main>
  <el-footer>
    <el-row>
      <el-col span="8" offset="8">
        <el-pagination current-page="1" page-sizes="10,20,30,40" page-size="10" layout="total, sizes, prev, pager, next, jumper" total="10" class="pagination"></el-pagination>
      </el-col>
    </el-row>
  </el-footer>
</el-container>
`;
